frame/bridge/ethereum/issuing/src/lib.rs

```diff
@@ -0,0 +1,419 @@
+			log::trace!(target: "darwinia-issuing", "start to register_or_issuing_erc20");
+			let user = ensure_signed(origin)?;
+			let tx_index = T::EthereumRelay::gen_receipt_index(&proof);
+			ensure!(!VerifiedIssuingProof::contains_key(tx_index), <Error<T>>::AssetAR);
+			let verified_receipt = T::EthereumRelay::verify_receipt(&proof)
+				.map_err(|err| {
+					log::trace!(target: "darwinia-issuing", "verify error {:?}", err);
+					<Error<T>>::ReceiptProofInv
+				})?;
+
+			let register_event = Abi::register_event();
+			let backing_event = Abi::backing_event();
+			let backing_address = EthereumBackingAddress::get();
+			let log_entry = verified_receipt
+				.logs
+				.into_iter()
+				.find(|x| {
+					x.address == backing_address &&
+						( x.topics[0] == register_event.signature()
+						  || x.topics[0] == backing_event.signature() )
+				})
+			.ok_or(<Error<T>>::LogEntryNE)?;
+
+			let input = if log_entry.topics[0] == register_event.signature() {
+				let ethlog = Self::parse_event(register_event, log_entry)?;
+				Self::process_erc20_creation(backing_address, ethlog)?
+			} else {
+				let ethlog = Self::parse_event(backing_event, log_entry)?;
+				Self::process_token_issuing(ethlog)?
+			};
+
+			let contract = MappingFactoryAddress::get();
+			let account = Self::dvm_account_id();
+			let basic = <T as darwinia_evm::Config>::AccountBasicMapping::account_basic(&account);
+			let substrate_account = <T as darwinia_evm::Config>::AddressMapping::into_account_id(account.clone());
+
+			<T as Config>::RingCurrency::transfer(&user, &substrate_account, T::FeeEstimate::get(), KeepAlive)?;
+			let transaction = Self::unsigned_transaction(basic.nonce, contract, input);
+			let result = dvm_ethereum::Module::<T>::do_transact(transaction, Some(account)).map_err(|e| -> &'static str {
+				log::trace!(target: "darwinia-issuing", "register_or_issuing_erc20 error {:?}", &e);
+				e.into()
+			} )?;
+
+			let leaved_balance = <T as Config>::RingCurrency::free_balance(&substrate_account);
+			// we should reserve some balance to keepalive this system account
+			let maxrefund = leaved_balance.saturating_sub(<T as Config>::RingCurrency::minimum_balance());
+
+			<T as Config>::RingCurrency::transfer(&substrate_account, &user, maxrefund, KeepAlive)?;
+			VerifiedIssuingProof::insert(tx_index, true);
+			Self::deposit_event(RawEvent::RegisteredOrRedeemed(user, backing_address));
+		}
+	}
+}
+
+impl<T: Config> Module<T> {
+	/// The account ID of the issuing pot.
+	pub fn dvm_account_id() -> H160 {
+		let account32: AccountId32 = T::ModuleId::get().into_account();
+		let account20: &[u8] = &account32.as_ref();
+		H160::from_slice(&account20[..20])
+	}
+
+	/// get dvm ethereum unsigned transaction
+	pub fn unsigned_transaction(
+		nonce: U256,
+		target: H160,
+		input: Vec<u8>,
+	) -> dvm_ethereum::Transaction {
+		dvm_ethereum::Transaction {
+			nonce,
+			gas_price: T::FeeCalculator::min_gas_price(),
```

Comment:
Could we overpass the minimum gas price check? @AsceticBear

Reply:
gas_price set to `None` to avoid gas fee problem

Reply:
`T::FeeCalculator::min_gas_price()` is not None actually?  Why not directly use None here, and require document here.

Reply:
The transaction defined as:

```rs
pub struct Transaction {
	pub nonce: U256,
	pub gas_price: U256,
	pub gas_limit: U256,
	pub action: TransactionAction,
	pub value: U256,
	pub input: Bytes,
	pub signature: TransactionSignature,
}
```
So, gas_price can not be None here. 

```rs
let (source, gas_price, nonce) = if need_signer {
	(
		Self::recover_signer(&transaction).ok_or_else(|| Error::<T>::InvalidSignature)?,
		Some(transaction.gas_price),
		Some(transaction.nonce),
	)
	} else {
		(H160::zero(), None, None)
        };
```
The final gas_price value will be None in `do_transact`, if I understand correctly.

Reply:
so, it's better to set gas_price as zero here.